service/code_interpreter.py

```diff
@@ -73,36 +82,65 @@ async def __aexit__(self, _exc_type, _exc_value, _traceback):
             self.sandbox.keep_alive(self.timeout)
         self.sandbox.close()
 
-    def get_files_code(self):
+    def get_dataframe(self):
         """
         Get the code to read the files in the sandbox.
         This can be used for instructing the LLM how to access the loaded files.
         """
-
         # TODO: Add support for xslx, json
-        files_code = "\n".join(
-            f'df{i} = pd.read_csv("{self._get_file_path(url)}") # {url}'
-            for i, url in enumerate(self.file_urls)
+        df = pd.read_csv(self.file_urls[0])
+        return df, self.file_urls[0]
+
+    def genreate_prompt(self, query: str) -> str:
+        df, url = self.get_dataframe()
+        return textwrap.dedent(
+            f"""
+        You are provided with a following pandas dataframe (`df`):
+        {df.info()}
+
+        Using the provided dataframe (`df`), update the following python code using pandas that returns the answer to question: \"{query}\"
+        
+        This is the initial python code to be updated:
+        
+        ```python
+        import pandas as pd
+
+        df = pd.read_csv("{url}") 
```

Comment:
I was thinking about this and maybe putting the public urls that can be used for downloading the data in the prompt may not be ok later on.
We can circumvent this by referring to the data with a different name, I just wanted to point this out from a privacy/security perspective.

Reply:
Correct, would you de-indentify on your end?